docs/releasing.md

```diff
@@ -0,0 +1,138 @@
+  `git push origin release-X.x` to create it
+- If creating a new patch release, use existing branch `release-X.x`:
+  `git checkout origin/release-X.x`
+- In the newly created release branch:
+   - Add a copy of upper-constraints.txt from [Openstack Requirements](https://opendev.org/openstack/requirements)
+     to replace the placeholder
+   - Pin ironic and ironic-inspector to match the corresponding bugfix or
+     stable branches
+   - If needed, pin versions of ironic-lib and sushy to grant compatibility with
+     ironic and ironic-inspector; this can probably be avoided considering that
+     we're using upper-constraints
+
+### Tags
+
+First we create a primary release tag, that triggers release note creation and
+image building processes.
+
+- Create a signed, annotated tag with: `git tag -s -a vX.x.y -m vX.x.y`
+- Push the tags to the GitHub repository: `git push origin vX.x.y`
+
+TODO(elfosardo): we should probably create an automated workflow with github
+actions as it's been done for other repositories like BMO
+
+### Release artifacts
+
+NOTE(elfosardo): TODO once we have an automated workflow
+
+### Release notes
+
+Next step is to clean up the release note manually.
+
+- Check for duplicates, reverts, and incorrect classifications of PRs, and
+  whatever release creation tagged to be manually checked.
+- For any superseded PRs (like same dependency uplifted multiple times, or
+  commit revertions) that provide no value to the release, move them to
+  Superseded section. This way the changes are acknowledged to be part of the
+  release, but not overwhelming the important changes contained by the release.
+- If the release you're making is not a new major release, new minor release, or
+  a new patch release from the latest release branch, uncheck the box for latest
+  release.
+- If it is a release candidate (RC) or a pre-release, tick pre-release box.
+- Publish the release.
+
+## Post-release actions for new release branches
+
+Some post-release actions are needed when a new release branch is created.
+
+### Branch protection rules
+
+Branch protection rules need to be applied to the new release branch. Copy the
+settings after the previous release branch, with the exception of
+`Required tests` selection. Required tests can only be selected after new
+keywords are implemented in Jenkins JJB, and project-infra, and have been run at
+least once in the PR targeting the branch in question.
+
+### Update README.md and build badges
+
+Update `README.md` with release specific information, both on `main` and in the
+new `release-X.x` branch as necessary.
+
+<!-- No example PR yet. To be added when first release from branch is made -->
+
+In the `release-X.x` branch, update the build badges in the `README.md` to point
+to correct Jenkins jobs, so the build statuses of the release branch are
+visible.
+
+<!-- No example PR yet. To be added when first release from branch is made -->
+
+### Branches lifecycle
```

Comment:
This section should probably be elsewhere, like `CONTRIBUTING.md`, like it is in CAPM3, IPAM and BMO.

Reply:
I can move it, but honestly it does make sense to me to keep it here since it's the main place where we talk about branches and release strategy

Reply:
Sure, no strong opinion. We can always move later if feeling like it.